libvips/foreign/vips2tiff.c

```diff
@@ -654,50 +654,52 @@ wtiff_write_header( Wtiff *wtiff, Layer *layer )
 		TIFFSetField( tif, TIFFTAG_BITSPERSAMPLE, 
 			vips_format_sizeof( wtiff->im->BandFmt ) << 3 );
 
-		if( wtiff->im->Bands < 3 ) {
-			/* Mono or mono + alpha.
-			 */
-			photometric = wtiff->miniswhite ? 
-				PHOTOMETRIC_MINISWHITE :  
+		if (wtiff->im->Type == VIPS_INTERPRETATION_B_W ||
+			wtiff->im->Type == VIPS_INTERPRETATION_GREY16) {
+			// Mono or mono + alpha.
+			photometric = wtiff->miniswhite ?
+				PHOTOMETRIC_MINISWHITE :
 				PHOTOMETRIC_MINISBLACK;
 			colour_bands = 1;
 		}
+		else if( wtiff->im->Type == VIPS_INTERPRETATION_LAB || 
+			wtiff->im->Type == VIPS_INTERPRETATION_LABS ) {
+			photometric = PHOTOMETRIC_CIELAB;
+			colour_bands = 3;
+		}
+		else if( wtiff->im->Type == VIPS_INTERPRETATION_CMYK &&
+			wtiff->im->Bands >= 4 ) {
+			photometric = PHOTOMETRIC_SEPARATED;
+			TIFFSetField( tif, 
+				TIFFTAG_INKSET, INKSET_CMYK );
+			colour_bands = 4;
+		}
+		else if (wtiff->im->Type == VIPS_INTERPRETATION_MULTIBAND ) {
+			photometric = PHOTOMETRIC_SEPARATED;
```

Comment:
I don't think you can assume MULTIBAND images are the same as TIFF MULTIINK. I often use 7 or 8 band TIFFs and I SEPARATED is not correct for them. I think we have to delete this case.